src/Migrate.php

```diff
@@ -0,0 +1,1394 @@
+                                    $schemaName,
+                                    $tableName
+                                ),
+                                sprintf(
+                                    '%s.%s.%s',
+                                    Helper::quoteIdentifier($database),
+                                    $schemaName,
+                                    Helper::quoteIdentifier($tableName)
+                                ),
+                                sprintf(
+                                    '%s.%s.%s',
+                                    Helper::quoteIdentifier($database),
+                                    Helper::quoteIdentifier($schemaName),
+                                    $tableName
+                                ),
+                                sprintf(
+                                    '%s.%s.%s',
+                                    Helper::quoteIdentifier($database),
+                                    Helper::quoteIdentifier($schemaName),
+                                    Helper::quoteIdentifier($tableName)
+                                ),
+                            ];
+                            return in_array($v['name'], $validSchema);
+                        }
+                    );
+                    $ownershipOnTable = array_filter($tableGrants, fn($v) => $v['privilege'] === 'OWNERSHIP');
+                    assert(count($ownershipOnTable) === 1);
+
+                    $ownershipOnTable = current($ownershipOnTable);
+
+                    self::assignSharePrivilegesToRole(
+                        $database,
+                        $ownershipOnTable['granted_by']
+                    );
+                    $this->destinationConnection->useRole($ownershipOnTable['granted_by']);
+                    $this->destinationConnection->useWarehouse($ownershipOnTable['granted_by']);
+
+                    if ($this->canCloneTable($database, $schemaName, $tableName)) {
+                        $this->logger->info(sprintf('Cloning table "%s" from OLD database', $tableName));
+                        $this->destinationConnection->query(sprintf(
+                            'CREATE TABLE %s.%s.%s CLONE %s.%s.%s;',
+                            Helper::quoteIdentifier($database),
+                            Helper::quoteIdentifier($schemaName),
+                            Helper::quoteIdentifier($tableName),
+                            Helper::quoteIdentifier($oldDbName),
+                            Helper::quoteIdentifier($schemaName),
+                            Helper::quoteIdentifier($tableName),
+                        ));
+                    } else {
+                        $this->logger->info(sprintf('Creating table "%s" from SHARE database', $tableName));
+                        $this->destinationConnection->query(sprintf(
+                            'CREATE TABLE %s.%s.%s AS SELECT * FROM %s.%s.%s;',
```

Comment:
Tohle bude problém u native datatypes :( 

Bude potřeba IMHO create table like table a pak insert into 

Problém CTAS je v tom, žeudělá všechny sloupce nullable.

Reply:
přepsal jsem to tady - https://github.com/keboola/project-migration-tool/pull/1/commits/e29e2d28bf0f201ff502cb736b49888e49c52711

Reply:
👍